### PR TITLE
Fix byte compiler warnings

### PR DESCRIPTION
**Pull request by phst**

### CODE REVIEW

File: puppet-mode.el [hunk 1]
```diff
@@ -987,7 +987,7 @@ Used as `syntax-propertize-function' in Puppet Mode."
 ;;; Dealing with strings
 (defun puppet-looking-around (back at)
   "Check if looking backwards at BACK and forward at AT."
-  (and (looking-at-p at) (looking-back back)))
+  (and (looking-at-p at) (looking-back back nil)))
```
Comment (bbatsov) [inline puppet-mode.el]: Looking back has just one required param - `(looking-back REGEXP &optional LIMIT GREEDY)`.

What's the warning you're trying to fix here?

Comment (phst) [inline puppet-mode.el]: It has two required parameters when taking the advertised calling convention into account, i.e. it's meant to have two required parameters, this is just not enforced by the interpreter (but the byte compiler warns nevertheless).

Comment (bbatsov) [inline puppet-mode.el]: That's pretty odd.

**Comment by bbatsov** [comment]
Thanks!

**Pull request merged**